src/NixFromNpm/Npm/Types.hs

```diff
@@ -25,14 +25,72 @@ data PackageInfo = PackageInfo {
   piTags :: Record SemVer
   } deriving (Show, Eq)
 
+-- | Taken from https://nodejs.org/api/process.html#process_process_platform,
+-- and filtered to those that correspond to nixpkgs platforms.
+data NodePlatform
+  = Darwin
+  | FreeBSD
+  | OpenBSD
+  | Linux
+  | SunOS
+  deriving (Show, Eq)
+
+-- | Convert a NodePlatform into text
+nodePlatformToText :: IsString t => NodePlatform -> t
+nodePlatformToText = \case
+  Darwin -> "darwin"
+  FreeBSD -> "freebsd"
+  OpenBSD -> "openbsd"
+  Linux -> "linux"
+  SunOS -> "solaris"
+
+-- | Parse a node platform from a string.
+parseNodePlatform :: Alternative f => Text -> f NodePlatform
+parseNodePlatform = \case
+  "linux" -> pure Linux
+  "darwin" -> pure Darwin
+  "freebsd" -> pure FreeBSD
+  "openbsd" -> pure OpenBSD
+  "sunos" -> pure SunOS
```

Comment:
In fact, I wonder if it'd be easier to change `SunOS` to `Solaris` and lowercase the `BSD` parts; then you could use the show and read instances with a toLower and an uppercase operation...

Reply:
So this is the way it is because node uses the term `sunos`, while the `lib.platforms` uses the term `solaris`.

This could definitely be clearer in the docs. And/Or we could use the `hnix` types to make it type safe.

Reply:
Oh! I understand.